File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;
        vector<vector<int>> freq(nums.size() + 1);

        for (const auto x : nums) {
            count[x] += 1;
        }

        for (const auto& [k, v] : count) {
            freq[v].push_back(k);
        }

        vector<int> res;
        for (int i = freq.size() - 1; i > 0; i--) {
            for (const auto num : freq[i]) {
                res.push_back(num);

                if (res.size() == k) {
                    return res;
                }
            }
        }
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        maxLength = 1
        seen = set()

        for num in nums:
            seen.add(num)

        for num in seen:
            if num - 1 not in seen:
                length = 1
                while (length + num) in seen:
                    length += 1

                maxLength = max(length, maxLength)

        return maxLength


        

        